core/src/readwriter.rs

```diff
@@ -39,6 +39,732 @@ impl ToSchema for &str {
+        }
+
+        if let Some(date_format) = &self.date_format {
+            options.insert("date_format".to_string(), date_format.clone());
+        }
+
+        if let Some(timestamp_format) = &self.timestamp_format {
+            options.insert("timestamp_format".to_string(), timestamp_format.clone());
+        }
+
+        if let Some(multi_line) = self.multi_line {
+            options.insert("multi_line".to_string(), multi_line.to_string());
+        }
+
+        if let Some(allow_unquoted_control_chars) = self.allow_unquoted_control_chars {
+            options.insert(
+                "allow_unquoted_control_chars".to_string(),
+                allow_unquoted_control_chars.to_string(),
+            );
+        }
+
+        if let Some(line_sep) = &self.line_sep {
+            options.insert("line_sep".to_string(), line_sep.clone());
+        }
+
+        if let Some(sampling_ratio) = self.sampling_ratio {
+            options.insert("sampling_ratio".to_string(), sampling_ratio.to_string());
+        }
+
+        if let Some(drop_field_if_all_null) = self.drop_field_if_all_null {
+            options.insert(
+                "drop_field_if_all_null".to_string(),
+                drop_field_if_all_null.to_string(),
+            );
+        }
+
+        if let Some(encoding) = &self.encoding {
+            options.insert("encoding".to_string(), encoding.clone());
+        }
+
+        if let Some(locale) = &self.locale {
+            options.insert("locale".to_string(), locale.clone());
+        }
+
+        if let Some(path_glob_filter) = self.path_glob_filter {
+            options.insert("path_glob_filter".to_string(), path_glob_filter.to_string());
+        }
+
+        if let Some(recursive_file_lookup) = self.recursive_file_lookup {
+            options.insert(
+                "recursive_file_lookup".to_string(),
+                recursive_file_lookup.to_string(),
+            );
+        }
+
+        if let Some(allow_non_numeric_numbers) = self.allow_non_numeric_numbers {
+            options.insert(
+                "allow_non_numeric_numbers".to_string(),
+                allow_non_numeric_numbers.to_string(),
+            );
+        }
+
+        options
+    }
+}
+
+/// A struct that represents options for configuring ORC file parsing.
+///
+/// # Options
+///
+/// - `merge_schema`: Merge schemas from all ORC files.
+/// - `path_glob_filter`: A glob pattern to filter files by their path.
+/// - `recursive_file_lookup`: Enable recursive search for files in directories.
+/// - `modified_before`: Only include files modified before a given timestamp.
+/// - `modified_after`: Only include files modified after a given timestamp.
+///
+/// # Example
+/// ```
+/// let options = OrcOptions::new()
+///     .merge_schema(true)
+///     .path_glob_filter("*.orc".to_string())
+///     .recursive_file_lookup(true);
+///
+/// let df = spark.read().orc("/path/to/orc", options)?;
+/// ```
+#[derive(Debug)]
+pub struct OrcOptions {
```

Comment:
can you also add in the `compression` option?

https://spark.apache.org/docs/latest/sql-data-sources-orc.html